src/Sentry.DiagnosticSource/Internals/DiagnosticSource/SentrySqlListener.cs

```diff
@@ -103,7 +103,7 @@ private void AddSpan(SentrySqlSpanType type, string operation, KeyValuePair<stri
                 }
 
                 if (type == SentrySqlSpanType.Execution &&
-                    value.GetProperty<Guid>(ConnectionKey) is { } connectionId)
+                    value.GetProperty<Guid>(ConnectionKey) is var connectionId)
```

Comment:
`is var` is just an assignment, not a null check.  But also `Guid` is a struct, so it will never be null.  You could assign and then compare to `default` or `Guid.Empty` if you that's a valid use case, but as-is, there's no point in it being in the `if` expression because it will always be true.

Reply:
I should be clearer.

If `Guid.Empty` is valid, then do:

```csharp
if (type == SentrySqlSpanType.Execution) {
  var connectionId = value.GetProperty<Guid>(ConnectionKey);
  // ...
```

If `Guid.Empty` is not valid, then do:

```csharp
if (type == SentrySqlSpanType.Execution &&
    value.GetProperty<Guid>(ConnectionKey) is var connectionId &&
    connectionId != default)
{
    // ...
```

Reply:
We can revisit this separately.  There's a few other parts of this file that have the same issue.  Thanks.